src/dendropy/datamodel/treemodel/_tree.py

```diff
@@ -3096,12 +3096,31 @@ def shuffle_taxa(self, include_internal_nodes=False, rng=None):
         assert len(current_to_shuffled_taxon_map) == len(current_node_taxon_map)
         return current_to_shuffled_taxon_map
 
-    def ladderize(self, ascending=True):
+    def ladderize(self, ascending=True, default_order=False):
         """
-        Sorts child nodes in ascending (if ``ascending`` is |False|) or
-        descending (if ``ascending`` is |False|) order in terms of the number of
-        children each child node has.
+        Sorts child nodes in ascending (if ``ascending`` is |True|) or descending (if
+        ``ascending`` is |False|) order in terms of the number of children each child
+        node has. If ``default_order`` is |True| then leaf nodes get sorted by taxon
+        labels when they are in polytomies.
         """
+
+        if default_order:
+            order = {
+                name: index
+                for index, name in enumerate(
+                    sorted(leaf.taxon.label for leaf in self.leaf_node_iter())
+                )
+            }
+
+        def dorder(node):  # -> int
+            if default_order:
+                try:
+                    return order[node.taxon.label]
+                except ValueError:
+                    return len(order)
+            else:
+                return 0
```

Comment:
This should be `return order.get(node.taxon.label, len(order)) if default_order else 0`.

Reply:
Actually, you could get rid of the `dorder` function, and just use

```python
key=lambda n: (node_desc_counts[n], order.get(node.taxon.label, len(order)) if default_order else 0)
```

in the `sort`.

Reply:
We can also just get rid of constructing the `order` dict and just pass the taxon label as the second item to sort on:

```python
key=lambda n: (
    node_desc_counts[n],
    n.taxon.label if default_order and hasattr(n.taxon, "label") else 0
),
```

(Also, the above naked except was catching cases where the taxon doesn't have a label, so added that here.

I guess that not all nodes must have a taxon, so there should also be `and hasattr(n, "taxon")` before checking the taxon has a label.